test/openssl/test_pkey_rsa.rb

```diff
@@ -46,63 +54,68 @@ def test_private
   end
 
   def test_new
-    key = OpenSSL::PKey::RSA.new(512)
-    assert_equal 512, key.n.num_bits
+    key = OpenSSL::PKey::RSA.new(2048)
+    assert_equal 2048, key.n.num_bits
     assert_equal 65537, key.e
     assert_not_nil key.d
 
     # Specify public exponent
-    key2 = OpenSSL::PKey::RSA.new(512, 3)
-    assert_equal 512, key2.n.num_bits
-    assert_equal 3, key2.e
+    key2 = OpenSSL::PKey::RSA.new(2048, 65537)
+    assert_equal 2048, key2.n.num_bits
+    assert_equal 65537, key2.e
     assert_not_nil key2.d
   end
 
   def test_s_generate
-    key1 = OpenSSL::PKey::RSA.generate(512)
-    assert_equal 512, key1.n.num_bits
+    key1 = OpenSSL::PKey::RSA.generate(2048)
+    assert_equal 2048, key1.n.num_bits
     assert_equal 65537, key1.e
 
     # Specify public exponent
-    key2 = OpenSSL::PKey::RSA.generate(512, 3)
-    assert_equal 512, key2.n.num_bits
-    assert_equal 3, key2.e
+    key2 = OpenSSL::PKey::RSA.generate(2048, 65537)
+    assert_equal 2048, key2.n.num_bits
+    assert_equal 65537, key2.e
     assert_not_nil key2.d
   end
 
   def test_new_break
-    assert_nil(OpenSSL::PKey::RSA.new(1024) { break })
+    assert_nil(OpenSSL::PKey::RSA.new(2048) { break })
     assert_raise(RuntimeError) do
-      OpenSSL::PKey::RSA.new(1024) { raise }
+      OpenSSL::PKey::RSA.new(2048) { raise }
     end
   end
 
   def test_sign_verify
-    rsa1024 = Fixtures.pkey("rsa1024")
+    # The ossl_rsa_check_key_size called in ossl_pkey_sign requires more than
+    # equal 2048 bits on the argument protect = 1 in FIPS.
+    # https://github.com/openssl/openssl/blob/3c6e11495975a4eda4cc5886080afed6203711ac/providers/common/securitycheck.c#L68-L69
```

Comment:
I think it may be actually verbose to explain it. This change seems fair without the FIPS requirement. IIRC, OpenSSL 1.1.1 introduced a minimum RSA key size in non-FIPS mode too, which required us to make some changes to the test suite.

Reply:
Sure. I will change the comment without using FIPS. Could you tell me the link to the reference document about the "OpenSSL 1.1.1 introduced a minimum RSA key size in non-FIPS mode too"?

Reply:
I think I misremembered. It apparently happened in 1.1.0. https://github.com/openssl/openssl/blob/a6c9378e43749cc75c47af0b7f1c9c9a1257e271/CHANGES.md#changes-between-102h-and-110--25-aug-2016

512-bit or 1024-bit RSA keys are not considered secure in any way, thus I think OpenSSL can drop support for it at any time. Attributing the limitation to FIPS 140 may be a red herring.

Reply:
I couldn't still find the part "512-bit or 1024-bit RSA keys are not considered secure in any way, thus I think OpenSSL can drop support for it at any time." in the `CHANGES.md`. Could you tell me the link with the "Code" tab?

https://github.com/openssl/openssl/blob/a6c9378e43749cc75c47af0b7f1c9c9a1257e271/CHANGES.md?plain=1#L4920

Reply:
It has an item "Increased the minimal RSA keysize from 256 to 512 bits"

I think it's fine to increase the key size used in this test case without an explanation because using 2048-bit keys makes sense in all aspects, regardless of the OpenSSL version or FIPS 140 requirements.

Reply:
Okay. So, the part is below.
https://github.com/openssl/openssl/blob/a6c9378e43749cc75c47af0b7f1c9c9a1257e271/CHANGES.md?plain=1#L5735-L5738

Reply:
On the latest rebase, there is no comment about the 2048-bits in the `test_sign_verify`.